FAILED src/integrity_tests/test_all.py::TestAllCatalogs::test_non_zero_cost

`test_non_zero_cost` fails today.

One of the providers returns 0 prices. 

```
Run pytest src/integrity_tests
============================= test session starts ==============================
platform linux -- Python 3.11.8, pytest-8.0.2, pluggy-1.4.0
rootdir: /home/runner/work/gpuhunt/gpuhunt
collected 25 items

src/integrity_tests/test_all.py F                                        [  4%]
src/integrity_tests/test_aws.py ...                                      [ 16%]
src/integrity_tests/test_azure.py .....                                  [ 36%]
src/integrity_tests/test_cudo.py F..                                     [ 48%]
src/integrity_tests/test_datacrunch.py ....                              [ 64%]
src/integrity_tests/test_gcp.py .....                                    [ 84%]
src/integrity_tests/test_nebius.py ....                                  [100%]

=================================== FAILURES ===================================
______________________ TestAllCatalogs.test_non_zero_cost ______________________

self = <integrity_tests.test_all.TestAllCatalogs object at 0x7f5729aaaa90>
catalog_files = [PosixPath('lambdalabs.csv'), PosixPath('cudo.csv'), PosixPath('nebius.csv'), PosixPath('gcp.csv'), PosixPath('aws.csv'), PosixPath('datacrunch.csv'), ...]

    def test_non_zero_cost(self, catalog_files: List[Path]):
        for file in catalog_files:
            with open(file, "r") as f:
                reader = csv.DictReader(f)
                prices = [float(row["price"]) for row in reader]
>           assert 0 not in prices
E           assert 0 not in [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, ...]

src/integrity_tests/test_all.py:19: AssertionError
________________________________ test_locations ________________________________

data_rows = [{'cpu': '1', 'disk_size': '', 'gpu_count': '1', 'gpu_memory': '16', ...}, {'cpu': '1', 'disk_size': '', 'gpu_count': ...u_count': '1', 'gpu_memory': '16', ...}, {'cpu': '3', 'disk_size': '', 'gpu_count': '1', 'gpu_memory': '16', ...}, ...]

    def test_locations(data_rows):
        expected = {
            "no-luster-1",
            "se-smedjebacken-1",
            "se-stockholm-1",
            "us-newyork-1",
            "us-santaclara-1",
        }
        locations = select_row(data_rows, "location")
>       assert set(locations) == expected
E       AssertionError: assert {'no-luster-1...santaclara-1'} == {'no-luster-1...santaclara-1'}
E         
E         Extra items in the right set:
E         'se-smedjebacken-1'
E         
E         Full diff:
E           {
E               'no-luster-1',
E         -     'se-smedjebacken-1',
E               'se-stockholm-1',
E               'us-newyork-1',
E               'us-santaclara-1',
E           }

src/integrity_tests/test_cudo.py:32: AssertionError
---------------------------- Captured stdout setup -----------------------------
.
=========================== short test summary info ============================
FAILED src/integrity_tests/test_all.py::TestAllCatalogs::test_non_zero_cost - assert 0 not in [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, ...]
FAILED src/integrity_tests/test_cudo.py::test_locations - AssertionError: assert {'no-luster-1...santaclara-1'} == {'no-luster-1...santaclara-1'}
  
  Extra items in the right set:
  'se-smedjebacken-1'
  
  Full diff:
    {
        'no-luster-1',
  -     'se-smedjebacken-1',
        'se-stockholm-1',
        'us-newyork-1',
        'us-santaclara-1',
    }
========================= 2 failed, 23 passed in 0.35s =========================
Error: Process completed with exit code 1.
```

--- a/.github/workflows/catalogs.yml
+++ b/.github/workflows/catalogs.yml
@@ -9,6 +9,11 @@ on:
   schedule:
     - cron: '0 1 * * *'  # 01:00 UTC every day
 
+env:
+  PIP_DISABLE_PIP_VERSION_CHECK: on
+  PIP_DEFAULT_TIMEOUT: 10
+  PIP_PROGRESS_BAR: off
+
 jobs:
   catalog-aws:
     name: Collect AWS catalog

--- a/src/gpuhunt/providers/azure.py
+++ b/src/gpuhunt/providers/azure.py
@@ -1,5 +1,6 @@
 import json
 import logging
+import math
 import os
 import re
 import time
@@ -146,10 +147,13 @@ class AzureProvider(AbstractProvider):
                     continue
                 if not item["armSkuName"]:
                     continue
+                price = float(item["retailPrice"])
+                if math.isclose(price, 0):
+                    continue
                 offer = RawCatalogItem(
                     instance_name=item["armSkuName"],
                     location=item["armRegionName"],
-                    price=item["retailPrice"],
+                    price=price,
                     spot="Spot" in item["meterName"],
                     cpu=None,
                     memory=None,

--- a/src/integrity_tests/test_all.py
+++ b/src/integrity_tests/test_all.py
@@ -1,19 +1,22 @@
 import csv
+import os
 from pathlib import Path
-from typing import List
 
 import pytest
 
+files = sorted(Path(os.environ["CATALOG_DIR"]).glob("*.csv"))
 
-@pytest.fixture
-def catalog_files(catalog_dir: Path) -> List[Path]:
-    return list(catalog_dir.glob("*.csv"))
+
+def catalog_name(catalog) -> str:
+    return catalog.name
 
 
 class TestAllCatalogs:
-    def test_non_zero_cost(self, catalog_files: List[Path]):
-        for file in catalog_files:
-            with open(file, "r") as f:
-                reader = csv.DictReader(f)
-                prices = [float(row["price"]) for row in reader]
-            assert 0 not in prices
+    @pytest.fixture(params=files, ids=catalog_name)
+    def catalog(self, request):
+        yield request.param
+
+    def test_non_zero_cost(self, catalog):
+        reader = csv.DictReader(catalog.open())
+        for row in reader:
+            assert float(row["price"]) != pytest.approx(0), str(row)

